src/gwt/src/org/rstudio/studio/client/workbench/copilot/Copilot.java

```diff
@@ -0,0 +1,356 @@
+                  GlobalDisplay globalDisplay,
+                  CopilotCommandBinder binder,
+                  CopilotServerOperations server)
+   {
+      display_ = display;
+      commands_ = commands;
+      sourceColumnManager_ = sourceColumnManager;
+      globalDisplay_ = globalDisplay;
+      server_ = server;
+      
+      binder.bind(commands, this);
+   }
+   
+   public void ensureAgentInstalled(CommandWithArg<Boolean> callback)
+   {
+      String progressLabel = "Verifying copilot installation...";
+      server_.copilotVerifyInstalled(
+            new DelayedProgressRequestCallback<CopilotVerifyInstalledResponse>(progressLabel)
+      {
+         @Override
+         protected void onSuccess(CopilotVerifyInstalledResponse response)
+         {
+            if (response.installed)
+            {
+               callback.execute(true);
+            }
+            else
+            {
+               installAgentWithPrompt(callback);
+            }
+         }
+      });
+   }
+   
+   private void installAgentWithPrompt(CommandWithArg<Boolean> callback)
+   {
+      CopilotInstallDialog dialog = new CopilotInstallDialog();
+      
+      dialog.addClickHandler(new ClickHandler()
+      {
+         @Override
+         public void onClick(ClickEvent event)
+         {
+            CommandWithArg<Boolean> wrappedCallback = (result) ->
+            {
+               dialog.closeDialog();
+               callback.execute(result);
+            };
+            
+            installAgent(
+                  dialog.getProgressIndicator(),
+                  wrappedCallback);
+         }
+      });
+      
+      dialog.addCancelHandler(new ClickHandler()
+      {
+         @Override
+         public void onClick(ClickEvent event)
+         {
+            dialog.closeDialog();
+            callback.execute(false);
+         }
+      });
+      
+      dialog.showModal();
+   }
+   
+   private void installAgent(ProgressIndicator indicator,
+                             CommandWithArg<Boolean> callback)
+   {
+      indicator.onProgress("Installing...");
+      server_.copilotInstallAgent(
+            new ServerRequestCallback<CopilotInstallAgentResponse>()
+            {
+               @Override
+               public void onResponseReceived(CopilotInstallAgentResponse response)
+               {
+                  indicator.onCompleted();
+                  
+                  String error = response.error;
+                  if (error != null)
+                  {
+                     display_.showErrorMessage(
+                           "An error occurred while installing GitHub Copilot.\n\n" +
+                           error);
+                  callback.execute(false);
+                  }
+                  else
+                  {
+                     display_.showMessage(
+                           MessageDisplay.MSG_INFO,
+                           "GitHub Copilot: Install Agent",
```

Comment:
GWT has the 'i18nCreator' command line tool to help here:

https://www.gwtproject.org/doc/latest/RefCommandLineTools.html

But it doesn't handle automatically externalizing strings to a property file. That, however, can be done through Eclipse, using Source -> Externalize Strings. So, I think the quickest way forward is:

1. Use Eclipse's Source -> Externalize Strings to create a property file,
2. Delete the generated Java interface,
3. Use the `i18nCreator` tool to auto-generate (or, auto-update) the GWT interface.

I'll see if I can wrap this up into some kind of helper so that we can easily and automatically update and generate these.